### DATA PREPROCESSING

In [2]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')
pd.set_option('display.max_columns', None)

In [3]:
#Example of How to Create DataFrame with Dictionary Variable 

#Dictionary 1,2,3:
PialaEropa = {"Negara" : ["jerman", "spanyol", "uni soviet", "italia", "prancis"], 
              "piala": [3, 3, 1, 1, 2] }
PialaAsia = {"Negara" : ["jepang", "indonesia", "korea selatan", "irak", "saudi arabia"], 
             "piala" : [4, 0, 2, 1, 3]}
PialaDunia = {"Negara" : ["jerman", "spanyol", "korea selatan", "jepang", "prancis"],
             "piala" : [4, 1, 0, 0, 3]}

#Membuat DataFrame dari Dictionary 1,2,3:
dfPialaEropa= pd.DataFrame.from_dict(PialaEropa)
dfPialaAsia= pd.DataFrame.from_dict(PialaAsia)
dfPialaDunia= pd.DataFrame.from_dict(PialaDunia)

In [4]:
dfPialaEropa

Negara  piala
0      jerman      3
1     spanyol      3
2  uni soviet      1
3      italia      1
4     prancis      2

In [5]:
dfPialaAsia

Negara  piala
0         jepang      4
1      indonesia      0
2  korea selatan      2
3           irak      1
4   saudi arabia      3

In [6]:
dfPialaDunia

Negara  piala
0         jerman      4
1        spanyol      1
2  korea selatan      0
3         jepang      0
4        prancis      3

In [7]:
#Example of How to Group Variables 
Pemenang = pd.concat([dfPialaEropa, dfPialaAsia, dfPialaDunia]).reset_index()
Pemenang

index         Negara  piala
0       0         jerman      3
1       1        spanyol      3
2       2     uni soviet      1
3       3         italia      1
4       4        prancis      2
5       0         jepang      4
6       1      indonesia      0
7       2  korea selatan      2
8       3           irak      1
9       4   saudi arabia      3
10      0         jerman      4
11      1        spanyol      1
12      2  korea selatan      0
13      3         jepang      0
14      4        prancis      3

In [8]:
#Example of How to Sum Up Row with the Same Variable
Total_Piala = Pemenang.groupby(['Negara']).sum()[['piala']]
Total_Piala.reset_index()

Negara  piala
0      indonesia      0
1           irak      1
2         italia      1
3         jepang      4
4         jerman      7
5  korea selatan      2
6        prancis      5
7   saudi arabia      3
8        spanyol      4
9     uni soviet      1

In [9]:
#Example of How to Sort Values, False to sort Z-A, True to sort A-Z
juaraBertahan = Total_Piala.sort_values(by='piala', ascending = False).reset_index()
juaraBertahan

Negara  piala
0         jerman      7
1        prancis      5
2         jepang      4
3        spanyol      4
4   saudi arabia      3
5  korea selatan      2
6           irak      1
7         italia      1
8     uni soviet      1
9      indonesia      0

In [10]:
#Example of How to Show Variable Index Value

JuaraBertahan = juaraBertahan.iloc[[0,3,5]].reset_index()
JuaraBertahan

index         Negara  piala
0      0         jerman      7
1      3        spanyol      4
2      5  korea selatan      2

In [11]:
#Example of How to Filter Conditional Data
Pemenang[(Pemenang['piala'] >= 2)]

index         Negara  piala
0       0         jerman      3
1       1        spanyol      3
4       4        prancis      2
5       0         jepang      4
7       2  korea selatan      2
9       4   saudi arabia      3
10      0         jerman      4
14      4        prancis      3

#### Example of Data Preprocessing Implementation

In [12]:
#Example of How to Import File

dfCreditScore = pd.read_csv(r"E:\MINE\beasiswa\2022\DATA BANGLORE\ON BOARDING\EXAM + TASK\3 - DATA PREPROCESSING\DATA TASK\credit_score.csv", sep = ',')
dfCreditScore

umur    pendapatan  family_member          debt  is_credit  \
0      60.854010  1.688747e+07            0.0  1.234051e+07        0.0   
1      65.603151  1.814031e+07            8.0  2.148741e+07        0.0   
2      27.203123  1.173253e+07            3.0  3.114519e+07        0.0   
3      34.861070  1.198259e+07            1.0  3.091853e+06        0.0   
4      51.776474  1.231265e+07            5.0  3.054250e+07        0.0   
...          ...           ...            ...           ...        ...   
13077  50.131497  1.053330e+07            1.0  1.750930e+07        0.0   
13078  56.835766  1.284579e+07            1.0  2.408513e+07        0.0   
13079  52.344909  1.430617e+07            0.0  2.958113e+07        0.0   
13080  47.782509  1.235632e+07            6.0  4.849752e+07        0.0   
13081  55.350772  1.351740e+07           12.0  2.548000e+07        0.0   

       credit_score  
0        514.435008  
1        507.763234  
2        454.941676  
3        437.284466  
4        470.543029  
...             ...  
13077    470.452534  
13078    498.080274  
13079    503.637435  
13080    478.337996  
13081    440.360202  

[13082 rows x 6 columns]

In [13]:
#Checking Data Type 
dfCreditScore.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13082 entries, 0 to 13081
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   umur           12941 non-null  float64
 1   pendapatan     12893 non-null  float64
 2   family_member  12915 non-null  float64
 3   debt           12887 non-null  float64
 4   is_credit      12919 non-null  float64
 5   credit_score   12920 non-null  float64
dtypes: float64(6)
memory usage: 613.3 KB


In [14]:
#Checking missing data
dfCreditScore.isnull().sum()

umur             141
pendapatan       189
family_member    167
debt             195
is_credit        163
credit_score     162
dtype: int64

In [15]:
#Checking statistical description of the data
dfCreditScore.describe()

umur    pendapatan  family_member          debt     is_credit  \
count  12941.000000  1.289300e+04   12915.000000  1.288700e+04  12919.000000   
mean      52.643255  1.388759e+07       2.528378  3.831354e+07      0.157907   
std        7.913492  2.742709e+06       2.843177  2.285632e+07      0.364668   
min       22.000000  7.540000e+06       0.000000  1.000000e+06      0.000000   
25%       47.345705  1.197522e+07       1.000000  2.233565e+07      0.000000   
50%       52.699454  1.344787e+07       2.000000  3.325222e+07      0.000000   
75%       58.004179  1.532839e+07       4.000000  4.884514e+07      0.000000   
max       80.000000  3.400000e+07      13.000000  2.340000e+08      1.000000   

       credit_score  
count  12920.000000  
mean     491.005186  
std       50.473637  
min      300.000000  
25%      462.060752  
50%      488.390956  
75%      518.246356  
max      850.000000

In [16]:
#Handle missing data 

dfCreditScore['umur']=dfCreditScore['umur'].fillna(dfCreditScore['umur'].median())
dfCreditScore['pendapatan']=dfCreditScore['pendapatan'].fillna(dfCreditScore['pendapatan'].median())
dfCreditScore['family_member']=dfCreditScore['family_member'].fillna(dfCreditScore['family_member'].median())
dfCreditScore.isnull().sum()

umur               0
pendapatan         0
family_member      0
debt             195
is_credit        163
credit_score     162
dtype: int64

In [17]:
#Changing Data Type of 'umur, family_member, pendapatan' 

dfCreditScore['umur'] = dfCreditScore['umur'].astype(int)
dfCreditScore['family_member'] = dfCreditScore['family_member'].astype(int)
dfCreditScore['pendapatan'] = dfCreditScore['pendapatan'].astype(int)
dfCreditScore

umur  pendapatan  family_member          debt  is_credit  credit_score
0        60    16887465              0  1.234051e+07        0.0    514.435008
1        65    18140313              8  2.148741e+07        0.0    507.763234
2        27    11732526              3  3.114519e+07        0.0    454.941676
3        34    11982587              1  3.091853e+06        0.0    437.284466
4        51    12312654              5  3.054250e+07        0.0    470.543029
...     ...         ...            ...           ...        ...           ...
13077    50    10533295              1  1.750930e+07        0.0    470.452534
13078    56    12845793              1  2.408513e+07        0.0    498.080274
13079    52    14306172              0  2.958113e+07        0.0    503.637435
13080    47    12356315              6  4.849752e+07        0.0    478.337996
13081    55    13517399             12  2.548000e+07        0.0    440.360202

[13082 rows x 6 columns]

In [18]:
#Filtering Conditional Data to Know Certain Customer Characteristics that Get Credit Access

Acc_Credit = dfCreditScore[~((dfCreditScore['umur'] > 40) & (dfCreditScore['family_member'] >= 3) & (dfCreditScore['pendapatan'] < 15000000))]
Acc_Credit

umur  pendapatan  family_member          debt  is_credit  credit_score
0        60    16887465              0  1.234051e+07        0.0    514.435008
1        65    18140313              8  2.148741e+07        0.0    507.763234
2        27    11732526              3  3.114519e+07        0.0    454.941676
3        34    11982587              1  3.091853e+06        0.0    437.284466
5        58    14380365              0  2.506403e+07        0.0    497.689723
...     ...         ...            ...           ...        ...           ...
13073    47    11412779              0  1.199996e+07        0.0    458.487993
13076    63    14627549              0  2.826608e+07        0.0    512.742323
13077    50    10533295              1  1.750930e+07        0.0    470.452534
13078    56    12845793              1  2.408513e+07        0.0    498.080274
13079    52    14306172              0  2.958113e+07        0.0    503.637435

[10199 rows x 6 columns]

In [19]:
#Filtering Conditional Data to Know Certain Customer Characteristics that Get Denied on Credit Access
Deny_PinjamanCS = dfCreditScore[((dfCreditScore['umur'] > 40) & (dfCreditScore['family_member'] >= 3) & (dfCreditScore['pendapatan'] < 15000000))]
Deny_PinjamanCS

umur  pendapatan  family_member          debt  is_credit  credit_score
4        51    12312654              5  3.054250e+07        0.0    470.543029
9        52    13008346              6  2.324459e+07        0.0    451.666608
25       59    12610213              8  3.559241e+07        1.0    388.288655
29       50    11017353              7  2.755120e+07        0.0    439.848123
32       49    10411430              3  4.791841e+07        0.0    486.677455
...     ...         ...            ...           ...        ...           ...
13070    45    13194456              4  5.744930e+07        0.0    497.888017
13074    47    12302117              4  4.429768e+07        1.0    396.391727
13075    46    14017970              5  2.664329e+07        1.0    373.082124
13080    47    12356315              6  4.849752e+07        0.0    478.337996
13081    55    13517399             12  2.548000e+07        0.0    440.360202

[2883 rows x 6 columns]

In [20]:
#Checking the correctness of the data, based on the amount of credit rejected and not rejected data
cekDataCS = pd.concat([Deny_PinjamanCS,Acc_Credit]).reset_index()
cekDataCS

index  umur  pendapatan  family_member          debt  is_credit  \
0          4    51    12312654              5  3.054250e+07        0.0   
1          9    52    13008346              6  2.324459e+07        0.0   
2         25    59    12610213              8  3.559241e+07        1.0   
3         29    50    11017353              7  2.755120e+07        0.0   
4         32    49    10411430              3  4.791841e+07        0.0   
...      ...   ...         ...            ...           ...        ...   
13077  13073    47    11412779              0  1.199996e+07        0.0   
13078  13076    63    14627549              0  2.826608e+07        0.0   
13079  13077    50    10533295              1  1.750930e+07        0.0   
13080  13078    56    12845793              1  2.408513e+07        0.0   
13081  13079    52    14306172              0  2.958113e+07        0.0   

       credit_score  
0        470.543029  
1        451.666608  
2        388.288655  
3        439.848123  
4        486.677455  
...             ...  
13077    458.487993  
13078    512.742323  
13079    470.452534  
13080    498.080274  
13081    503.637435  

[13082 rows x 7 columns]

In [21]:
#Checking Statistical Description of Customer that Get Credit Access
Acc_Credit.describe()

umur    pendapatan  family_member          debt     is_credit  \
count  10199.000000  1.019900e+04   10199.000000  1.005000e+04  10065.000000   
mean      52.695166  1.423038e+07       1.636239  3.832340e+07      0.154496   
std        8.413649  2.905905e+06       2.156380  2.284538e+07      0.361441   
min       22.000000  7.540000e+06       0.000000  1.000000e+06      0.000000   
25%       47.000000  1.213576e+07       0.000000  2.223035e+07      0.000000   
50%       53.000000  1.379009e+07       1.000000  3.330365e+07      0.000000   
75%       59.000000  1.592604e+07       2.000000  4.888313e+07      0.000000   
max       80.000000  3.400000e+07      13.000000  2.340000e+08      1.000000   

       credit_score  
count  10063.000000  
mean     499.596030  
std       49.073392  
min      345.856695  
25%      471.275404  
50%      496.179995  
75%      525.878105  
max      850.000000

Hasil Analisis:
1. Rata-rata umur pelanggan yang tidak ditolak pengajuan pinjamannya adalah 53 tahun
2. Rata-rata pendapatan pelanggan yang tidak ditolak pengajuan pinjamannya adalah 14.230.380
3. Rata-rata jumlah family member pelanggan yang tidak ditolak pengajuan pinjamannya adalah 2 family member

In [23]:
#Import file data banking
df_Cbanking = pd.read_csv(r"E:\MINE\beasiswa\2022\DATA BANGLORE\ON BOARDING\EXAM + TASK\3 - DATA PREPROCESSING\DATA TASK\banking.csv")
df_Cbanking

age          job  marital            education  default housing loan  \
0       44  blue-collar  married             basic.4y  unknown     yes   no   
1       53   technician  married              unknown       no      no   no   
2       28   management   single    university.degree       no     yes   no   
3       39     services  married          high.school       no      no   no   
4       55      retired  married             basic.4y       no     yes   no   
...    ...          ...      ...                  ...      ...     ...  ...   
41183   59      retired  married          high.school  unknown      no  yes   
41184   31    housemaid  married             basic.4y  unknown      no   no   
41185   42       admin.   single    university.degree  unknown     yes  yes   
41186   48   technician  married  professional.course       no      no  yes   
41187   25      student   single          high.school       no      no   no   

         contact month day_of_week  duration  campaign  pdays  previous  \
0       cellular   aug         thu       210         1    999         0   
1       cellular   nov         fri       138         1    999         0   
2       cellular   jun         thu       339         3      6         2   
3       cellular   apr         fri       185         2    999         0   
4       cellular   aug         fri       137         1      3         1   
...          ...   ...         ...       ...       ...    ...       ...   
41183  telephone   jun         thu       222         1    999         0   
41184  telephone   may         thu       196         2    999         0   
41185  telephone   may         wed        62         3    999         0   
41186  telephone   oct         tue       200         2    999         0   
41187  telephone   may         fri       112         4    999         0   

          poutcome  emp_var_rate  cons_price_idx  cons_conf_idx  euribor3m  \
0      nonexistent           1.4          93.444          -36.1      4.963   
1      nonexistent          -0.1          93.200          -42.0      4.021   
2          success          -1.7          94.055          -39.8      0.729   
3      nonexistent          -1.8          93.075          -47.1      1.405   
4          success          -2.9          92.201          -31.4      0.869   
...            ...           ...             ...            ...        ...   
41183  nonexistent           1.4          94.465          -41.8      4.866   
41184  nonexistent           1.1          93.994          -36.4      4.860   
41185  nonexistent           1.1          93.994          -36.4      4.857   
41186  nonexistent          -3.4          92.431          -26.9      0.742   
41187  nonexistent           1.1          93.994          -36.4      4.859   

       nr_employed  y  
0           5228.1  0  
1           5195.8  0  
2           4991.6  1  
3           5099.1  0  
4           5076.2  1  
...            ... ..  
41183       5228.1  0  
41184       5191.0  0  
41185       5191.0  0  
41186       5017.5  0  
41187       5191.0  0  

[41188 rows x 21 columns]

In [24]:
#Checking Data Type
df_Cbanking.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 41188 entries, 0 to 41187
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   age             41188 non-null  int64  
 1   job             41188 non-null  object 
 2   marital         41188 non-null  object 
 3   education       41188 non-null  object 
 4   default         41188 non-null  object 
 5   housing         41188 non-null  object 
 6   loan            41188 non-null  object 
 7   contact         41188 non-null  object 
 8   month           41188 non-null  object 
 9   day_of_week     41188 non-null  object 
 10  duration        41188 non-null  int64  
 11  campaign        41188 non-null  int64  
 12  pdays           41188 non-null  int64  
 13  previous        41188 non-null  int64  
 14  poutcome        41188 non-null  object 
 15  emp_var_rate    41188 non-null  float64
 16  cons_price_idx  41188 non-null  float64
 17  cons_conf_idx   41188 non-null 

In [25]:
#checking missing data 
df_Cbanking.isnull().sum()

age               0
job               0
marital           0
education         0
default           0
housing           0
loan              0
contact           0
month             0
day_of_week       0
duration          0
campaign          0
pdays             0
previous          0
poutcome          0
emp_var_rate      0
cons_price_idx    0
cons_conf_idx     0
euribor3m         0
nr_employed       0
y                 0
dtype: int64

In [26]:
#Checking statistical description of data 
df_Cbanking.describe(include='all')

age     job  marital          education default housing  \
count   41188.00000   41188    41188              41188   41188   41188   
unique          NaN      12        4                  8       3       3   
top             NaN  admin.  married  university.degree      no     yes   
freq            NaN   10422    24928              12168   32588   21576   
mean       40.02406     NaN      NaN                NaN     NaN     NaN   
std        10.42125     NaN      NaN                NaN     NaN     NaN   
min        17.00000     NaN      NaN                NaN     NaN     NaN   
25%        32.00000     NaN      NaN                NaN     NaN     NaN   
50%        38.00000     NaN      NaN                NaN     NaN     NaN   
75%        47.00000     NaN      NaN                NaN     NaN     NaN   
max        98.00000     NaN      NaN                NaN     NaN     NaN   

         loan   contact  month day_of_week      duration      campaign  \
count   41188     41188  41188       41188  41188.000000  41188.000000   
unique      3         2     10           5           NaN           NaN   
top        no  cellular    may         thu           NaN           NaN   
freq    33950     26144  13769        8623           NaN           NaN   
mean      NaN       NaN    NaN         NaN    258.285010      2.567593   
std       NaN       NaN    NaN         NaN    259.279249      2.770014   
min       NaN       NaN    NaN         NaN      0.000000      1.000000   
25%       NaN       NaN    NaN         NaN    102.000000      1.000000   
50%       NaN       NaN    NaN         NaN    180.000000      2.000000   
75%       NaN       NaN    NaN         NaN    319.000000      3.000000   
max       NaN       NaN    NaN         NaN   4918.000000     56.000000   

               pdays      previous     poutcome  emp_var_rate  cons_price_idx  \
count   41188.000000  41188.000000        41188  41188.000000    41188.000000   
unique           NaN           NaN            3           NaN             NaN   
top              NaN           NaN  nonexistent           NaN             NaN   
freq             NaN           NaN        35563           NaN             NaN   
mean      962.475454      0.172963          NaN      0.081886       93.575664   
std       186.910907      0.494901          NaN      1.570960        0.578840   
min         0.000000      0.000000          NaN     -3.400000       92.201000   
25%       999.000000      0.000000          NaN     -1.800000       93.075000   
50%       999.000000      0.000000          NaN      1.100000       93.749000   
75%       999.000000      0.000000          NaN      1.400000       93.994000   
max       999.000000      7.000000          NaN      1.400000       94.767000   

        cons_conf_idx     euribor3m   nr_employed             y  
count    41188.000000  41188.000000  41188.000000  41188.000000  
unique            NaN           NaN           NaN           NaN  
top               NaN           NaN           NaN           NaN  
freq              NaN           NaN           NaN           NaN  
mean       -40.502600      3.621291   5167.035911      0.112654  
std          4.628198      1.734447     72.251528      0.316173  
min        -50.800000      0.634000   4963.600000      0.000000  
25%        -42.700000      1.344000   5099.100000      0.000000  
50%        -41.800000      4.857000   5191.000000      0.000000  
75%        -36.400000      4.961000   5228.100000      0.000000  
max        -26.900000      5.045000   5228.100000      1.000000

In [27]:
#Checking categorical variables in marital column
df_Cbanking['marital'].value_counts()

married     24928
single      11568
divorced     4612
unknown        80
Name: marital, dtype: int64

In [28]:
#Checking categorical variables in housing column
df_Cbanking['housing'].value_counts()

yes        21576
no         18622
unknown      990
Name: housing, dtype: int64

In [29]:
#Filtering conditional banking data
DenyPinjaman = df_Cbanking[(df_Cbanking['marital'] != 'married') & (df_Cbanking['housing'] == 'no')]
DenyPinjaman

age          job   marital            education  default housing loan  \
10      34    housemaid    single    university.degree       no      no   no   
17      27       admin.    single    university.degree       no      no   no   
21      35  blue-collar    single              unknown       no      no  yes   
23      32   technician    single  professional.course       no      no   no   
29      38  blue-collar    single             basic.4y  unknown      no   no   
...    ...          ...       ...                  ...      ...     ...  ...   
41154   31   technician  divorced          high.school       no      no   no   
41156   44   technician  divorced  professional.course       no      no   no   
41159   47       admin.  divorced          high.school  unknown      no   no   
41167   30       admin.    single          high.school       no      no   no   
41187   25      student    single          high.school       no      no   no   

         contact month day_of_week  duration  campaign  pdays  previous  \
10     telephone   may         fri        62         2    999         0   
17     telephone   oct         tue       540         1    999         0   
21     telephone   jun         fri      1114         1    999         0   
23      cellular   jul         thu        35         1    999         0   
29     telephone   jul         tue        41         5    999         0   
...          ...   ...         ...       ...       ...    ...       ...   
41154   cellular   aug         fri       101         4    999         0   
41156  telephone   jun         thu       112         1    999         0   
41159   cellular   jul         mon       129         4    999         0   
41167   cellular   jul         wed       251         3    999         0   
41187  telephone   may         fri       112         4    999         0   

          poutcome  emp_var_rate  cons_price_idx  cons_conf_idx  euribor3m  \
10     nonexistent           1.1          93.994          -36.4      4.864   
17     nonexistent          -0.1          93.798          -40.4      4.860   
21     nonexistent           1.4          94.465          -41.8      4.967   
23     nonexistent           1.4          93.918          -42.7      4.968   
29     nonexistent           1.4          93.918          -42.7      4.961   
...            ...           ...             ...            ...        ...   
41154  nonexistent           1.4          93.444          -36.1      4.964   
41156  nonexistent           1.4          94.465          -41.8      4.958   
41159  nonexistent           1.4          93.918          -42.7      4.960   
41167  nonexistent           1.4          93.918          -42.7      4.963   
41187  nonexistent           1.1          93.994          -36.4      4.859   

       nr_employed  y  
10          5191.0  0  
17          5195.8  1  
21          5228.1  0  
23          5228.1  0  
29          5228.1  0  
...            ... ..  
41154       5228.1  0  
41156       5228.1  0  
41159       5228.1  0  
41167       5228.1  0  
41187       5191.0  0  

[7233 rows x 21 columns]

In [30]:
#Filtering banking data based on analysis parameter 
Deny_Pinjaman = DenyPinjaman[['age','duration','marital','housing']]
Deny_Pinjaman

age  duration   marital housing
10      34        62    single      no
17      27       540    single      no
21      35      1114    single      no
23      32        35    single      no
29      38        41    single      no
...    ...       ...       ...     ...
41154   31       101  divorced      no
41156   44       112  divorced      no
41159   47       129  divorced      no
41167   30       251    single      no
41187   25       112    single      no

[7233 rows x 4 columns]

In [31]:
#Calculate the Average of 'age'
Deny_Pinjaman['age'].mean()

36.646757915111294

In [32]:
#Processing duration data
Deny_Pinjaman['duration_inYear'] = Deny_Pinjaman['duration'] / 365
Deny_Pinjaman

age  duration   marital housing  duration_inYear
10      34        62    single      no         0.169863
17      27       540    single      no         1.479452
21      35      1114    single      no         3.052055
23      32        35    single      no         0.095890
29      38        41    single      no         0.112329
...    ...       ...       ...     ...              ...
41154   31       101  divorced      no         0.276712
41156   44       112  divorced      no         0.306849
41159   47       129  divorced      no         0.353425
41167   30       251    single      no         0.687671
41187   25       112    single      no         0.306849

[7233 rows x 5 columns]

In [33]:
#Sorting loan data based on duration
Deny_Pinjaman.sort_values("duration", ascending = False)

age  duration   marital housing  duration_inYear
32880   27      3785    single      no        10.369863
37790   35      2680    single      no         7.342466
35742   26      2635    single      no         7.219178
24883   53      2486  divorced      no         6.810959
11155   33      2420    single      no         6.630137
...    ...       ...       ...     ...              ...
39614   30         5    single      no         0.013699
27404   27         5    single      no         0.013699
39692   25         5    single      no         0.013699
3738    26         4    single      no         0.010959
39151   34         3    single      no         0.008219

[7233 rows x 5 columns]

In [34]:
#Checking Statistical Description Based On Filtered Data
Deny_Pinjaman.describe()

age     duration  duration_inYear
count  7233.000000  7233.000000      7233.000000
mean     36.646758   261.425411         0.716234
std       9.950239   257.478976         0.705422
min      17.000000     3.000000         0.008219
25%      30.000000   104.000000         0.284932
50%      34.000000   184.000000         0.504110
75%      42.000000   327.000000         0.895890
max      95.000000  3785.000000        10.369863

Hasil Analisis:
1. Pelanggan yang ditolak pinjamannya rata-rata berumur 37 tahun 
2. Durasi pinjaman terlama yakni 3785 hari atau 10 tahun
3. Berdasarkan data  yang telah difilter, rata-rata peminjam yang ditolak berumur 37 tahun dengan rata-rata durasi pinjaman
   selama 261 hari atau hampir dari satu tahun. 